crates/swc_ecma_transforms_compat/src/es2022/private_in_object.rs

```diff
@@ -367,6 +368,15 @@ impl VisitMut for PrivateInObject {
                 });
             }
 
+            Expr::Member(MemberExpr {
+                span,
+                obj: _,
+                prop: MemberProp::PrivateName(PrivateName { .. }),
+            }) => {
+                let error = "Illegal access to private class property.";
+                HANDLER.with(|h| h.struct_span_err(*span, error).emit());
+            }
+
             _ => {}
```

Comment:
Can you add some tests?

Reply:
You can add some `input.js` within proper directory in https://github.com/swc-project/swc/tree/5122a7dc6270cfa3fb44dbd32cf8e33b808eed0e/crates/swc/tests/errors

Reply:
And I think we need more integration tests (in `swc` crate) for valid cases, too.

Reply:
Will add tests.